### DISCUSSION TITLE
Using .npmrc reports packages as undefined, where to go next?

### DISCUSSION THREAD

**Discussion by gregoryduckworth** [renovatebot/renovate#8392] [Request Help] [answered]
I've been following the documentation at https://docs.renovatebot.com/private-modules/#when-are-npm-private-modules-used where I am `running your own Renovate bot: copy an .npmrc file to the home dir of the bot and it will work for all repositories`.

The `.npmrc` config is (as suggest by [npm](https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow#create-and-check-in-a-project-specific-npmrc-file))
```
//registry.npmjs.org/:_authToken=${NPM_TOKEN}
```

My `printenv` reports the `NPM_TOKEN=<token>` 

I've tried in the directory of the Renovate instance and at the user root directory running Renovate however it always fails.

Is there something additional I need to do or mixed up the steps?

Thanks

**Comment by gregoryduckworth** [comment, marked as answer]
Think this is something to do with how I am using docker.

Running the command `npm config set '//registry.npmjs.org/:_authToken' "${NPM_TOKEN}"` in the Dockerfile shows the problem however running it within docker works.

**Reply by rarkins** [reply]
Elaborate on "always fails". Where/how? You may also need to set trustLevel=high

**Reply by gregoryduckworth** [reply]
Sorry, when I mean always fail I get the error that it couldn't find the private npm repository because it cannot use the access token. As mentioned above, I think it is my dockerfile configuration that is wrong and not renovate itself.

**Reply by rarkins** [reply]
Can you try configuring the npmrc field in config.js or in env using RENOVATE_NPMRC?

**Reply by gregoryduckworth** [reply]
> Can you try configuring the npmrc field in config.js or in env using RENOVATE_NPMRC?

I tried both the `npmToken` (as I'm using private repositories from npm itself) and `RENOVATE_NPMRC` but both failed to access the private packages.

But using `npmrc: "//registry.npmjs.org/:_authToken=<TOKEN>"` in the `config.js` file did work.

**Reply by viceice** [reply]
Can you try to use a host rule with `npm` type instead?
